core/shared/platform/linux-sgx/sgx_platform.c

```diff
@@ -60,16 +60,18 @@ os_set_print_function(os_print_function_t pf)
 int
 os_printf(const char *message, ...)
 {
+    int bytes_written = 0;
+
     if (print_function != NULL) {
         char msg[FIXED_BUFFER_SIZE] = { '\0' };
         va_list ap;
         va_start(ap, message);
-        vsnprintf(msg, FIXED_BUFFER_SIZE, message, ap);
+        bytes_written += vsnprintf(msg, FIXED_BUFFER_SIZE, message, ap);
```

Comment:
It seems work, but should not the best way, we had better get the exact size from print_function. Some suggestions:
1. In platform_internal.h, change return type of os_print_function_t to int

2. Here, change L71 to `bytes_written += print_function(msg)`
    And should also apply same changes for os_vprintf in this file.

3. Had better enlarge macro FIXED_BUFFER_SIZE in this file, it is `1 << 9`, or 512, not sure whether it is enough for most cases? How about 4096?

4. Modify file `product-mini/platforms/linux-sgx/enclave-sample/Enclave/Enclave.edl`
    and `product-mini/platforms/linux-sgx/enclave-sample/Enclave/Enclave_minimal.edl`
    Change `void ocall_print([in, string]const char* str);` to `int ocall_print([in, string]const char* str);`

5. Modify `product-mini/platforms/linux-sgx/enclave-sample/App/App.cpp`, the implementation of ocall_print:
```C
int
ocall_print(const char *str)
{
    return printf("%s", str);
}
```